internal/configs/configload/loader_load.go

```diff
@@ -61,7 +61,7 @@ func (l *Loader) moduleWalkerLoad(req *configs.ModuleRequest) (*configs.Module,
 	var diags hcl.Diagnostics
 
 	// Check for inconsistencies between manifest and config
-	if req.SourceAddr.String() != record.SourceAddr {
+	if req.SourceAddr != nil && req.SourceAddr.String() != record.SourceAddr {
```

Comment:
Perhaps it would be worth noting here in what situation `req.SourceAddr` might be nil, and why it's therefore okay to just quietly skip this check in that case. (I assume it's because we should already have emitted some other error upstream saying that the `source` argument was invalid.)

Reply:
Good point, done!